gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/discord/ListenerService.java

```diff
@@ -201,4 +200,99 @@ public String getTheQuestion(SlashCommandInteractionEvent event) {
     }
 
 
-}
+    public List<DocumentInstanceSection> findClosestSectionWithHttpRequest(String token, Message message, UUID projectId) throws GendoxException {
+        List<DocumentInstanceSection> sectionList = new ArrayList<>();
+        // Prepare the HTTP headers with the bearer token
+        HttpHeaders headers = new HttpHeaders();
+        headers.setBearerAuth(token);
+
+
+        // Prepare the request entity with the message and headers
+        HttpEntity<Message> requestEntity = new HttpEntity<>(message, headers);
+
+        // Create a RestTemplate
+
+        String url = "http://localhost:8080/gendox/api/v1/messages/semantic-search?projectId=" + projectId.toString() + "&page=0&size=5";
+
+
+        ResponseEntity<List<LinkedHashMap>> responseEntity;
+        try {
+            ParameterizedTypeReference<List<LinkedHashMap>> responseType = new ParameterizedTypeReference<List<LinkedHashMap>>() {
+            };
+            responseEntity = restTemplate.exchange(url, HttpMethod.POST, requestEntity, responseType);
+        } catch (RestClientException e) {
+            // Handle any exceptions that occur during the request.
+            logger.error("Error sending HTTP request: " + e.getMessage(), e);
+            throw new GendoxException("ERROR_SENDING_REQUEST", "Error sending HTTP request", HttpStatus.BAD_REQUEST);
+        }
+
+        // Check if the request was successful
+        if (responseEntity.getStatusCode() == HttpStatus.OK) {
+            List<LinkedHashMap> linkedHashMapList = responseEntity.getBody();
+
+
+            // Iterate through the LinkedHashMap objects and extract values
+            for (LinkedHashMap linkedHashMap : linkedHashMapList) {
+                // Create a DocumentInstanceSection and set its properties from the LinkedHashMap
+                DocumentInstanceSection section = new DocumentInstanceSection();
+                // Convert the String ID to UUID
+                String idString = (String) linkedHashMap.get("id");
+                UUID id = UUID.fromString(idString);
+
+                section.setId(id);
+                section.setSectionValue((String) linkedHashMap.get("sectionValue")); // Replace "name" with the actual key
+
+                sectionList.add(section);
```

Comment:
This is too much boiler-plate code. the deserialization needs to be done in the restTemplate.

If this doesn't work, try to create a record like:
```
public record SemanticSearchResponce(List<DocumentInstanceSection> sections) {
    }
```

I would like all this code to be in an [Adapter class](https://www.digitalocean.com/community/tutorials/adapter-design-pattern-java) or using [Spring Rest Client](https://www.youtube.com/watch?v=UDNrJAvKc0k&themeRefresh=1) (dont sent more than 2 hours though).